File: sdk/java/src/main/java/com/learnerview/sab/model/ApiResponse.java
package com.learnerview.sab.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Generic API response wrapper.
 * @param <T> the type of the data field
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ApiResponse<T> {
    /**
     * Whether the operation was successful.
     */
    private boolean success;

    /**
     * Response message.
     */
    private String message;

    /**
     * Response data.
     */
    private T data;

    /**
     * Error code (if applicable).
     */
    private String errorCode;

    /**
     * Timestamp of the response.
     */
    private long timestamp;
}

File: sdk/nodejs/src/models/api-response.ts
import { ApiResponse } from './common';
//Generic API response wrapper.

export { ApiResponse };
//Helper function to check if response is successful.

export function isResponseSuccess<T>(response: ApiResponse<T>): boolean {
  return response.success === true;
}
//Helper function to check if response has data.

export function hasResponseData<T>(response: ApiResponse<T>): boolean {
  return response.data !== undefined && response.data !== null;
}
//Helper function to extract data from response.

export function getResponseData<T>(response: ApiResponse<T>): T | undefined {
  return response.data;
}
//Helper function to extract error information from response.

export function getResponseError<T>(response: ApiResponse<T>): {
  message?: string;
  errorCode?: string;
} {
  return {
    message: response.message,
    errorCode: response.errorCode,
  };
}
//Helper function to create a successful response.

export function createSuccessResponse<T>(data: T, message?: string): ApiResponse<T> {
  return {
    success: true,
    message,
    data,
    timestamp: Date.now(),
  };
}
//Helper function to create an error response.

export function createErrorResponse(message: string, errorCode?: string, data?: any): ApiResponse {
  return {
    success: false,
    message,
    errorCode,
    data,
    timestamp: Date.now(),
  };
}
